File: android/app/src/main/kotlin/com/ventechsolutions/myFHB/MyApp.java
package com.ventechsolutions.myFHB;

import android.app.Activity;
import android.app.Application;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.content.ContentResolver;
import android.content.Context;
import android.content.Intent;
import android.media.RingtoneManager;
import android.net.Uri;
import android.os.Build;

import androidx.annotation.CallSuper;
import androidx.core.app.NotificationManagerCompat;

import com.ventechsolutions.myFHB.services.OnTapNotification;

import java.util.Map;

import io.flutter.view.FlutterMain;

public class MyApp extends Application {
    @Override
    @CallSuper
    public void onCreate() {
        super.onCreate();
        FlutterMain.startInitialization(this);
    }

    public static boolean isMissedNSShown=true;

    private Activity mCurrentActivity = null;

    public Activity getCurrentActivity() {
        return mCurrentActivity;
    }

    public void setCurrentActivity(Activity mCurrentActivity) {
        this.mCurrentActivity = mCurrentActivity;
    }
}

File: android/app/src/main/kotlin/com/ventechsolutions/myFHB/MyApp.kt
package com.ventechsolutions.myFHB

import android.app.Activity
import android.app.Application
import android.os.Bundle
import androidx.annotation.CallSuper
import com.google.firebase.firestore.FirebaseFirestore
import com.lifesense.plugin.ble.LSBluetoothManager
import io.flutter.view.FlutterMain
import java.lang.ref.WeakReference

public class MyApp : Application(), Application.ActivityLifecycleCallbacks {
    @CallSuper
    override fun onCreate() {
        super.onCreate()
//        OHQDeviceManager.init(applicationContext,currentActivity)
        registerActivityLifecycleCallbacks(this)
        FlutterMain.startInitialization(this)
        LSBluetoothManager.getInstance().initManager(getApplicationContext())
        LSBluetoothManager.getInstance().registerBluetoothReceiver(getApplicationContext());
        LSBluetoothManager.getInstance().openDebugMode("LifesenseBluetooth")


    }

    var currentActivity: Activity? = null

    companion object {
        var foregroundActivityRef: WeakReference<Activity>? = null
        var isMissedNSShown = true
        var recordId = ""
        const val TAG: String = "MyApp"
    }

    fun updateStatus(isAccept: Boolean) {
        var data: HashMap<String, Any>
        if (isAccept) {
            data = hashMapOf(
                    "call_status" to "accept"
            )
        } else {
            data = hashMapOf(
                    "call_status" to "decline"
            )
        }

        try {
            var db = FirebaseFirestore.getInstance()
            db.collection("call_log")
                    .document(recordId)
                    .update(data)
        } catch (e: Exception) {
            print(e.message)
        }
    }

    override fun onActivityCreated(p0: Activity, p1: Bundle?) {
    }

    override fun onActivityStarted(activity: Activity) {
        foregroundActivityRef = WeakReference(activity)
    }

    override fun onActivityResumed(activity: Activity) {
        foregroundActivityRef = WeakReference(activity)
    }

    override fun onActivityPaused(activity: Activity) {
        if (foregroundActivityRef != null && foregroundActivityRef?.get() == activity) {
            foregroundActivityRef = null;
        }
    }

    override fun onActivityStopped(activity: Activity) {
        if (foregroundActivityRef != null && foregroundActivityRef?.get() == activity) {
            foregroundActivityRef = null;
        }
    }

    override fun onActivitySaveInstanceState(p0: Activity, p1: Bundle) {

    }

    override fun onActivityDestroyed(p0: Activity) {

    }


}

